app/src/main/java/com/pedrodavidlp/ittrivial/Roulette.kt

```diff
@@ -1,198 +1,89 @@
+    animation.setAnimationListener(object : Animation.AnimationListener {
+      override fun onAnimationRepeat(p0: Animation?) {
 
-   * @param newTopDiv
-   */
-  fun setAlternateTopDiv(newTopDiv: Int) {
+      }
 
-    if (newTopDiv < 0 || newTopDiv >= divCount)
-      return
-    else
-      rTop = newTopDiv
+      override fun onAnimationEnd(p0: Animation?) {
+        rChangeListener.onSelectionChange(current)
+        rotation += auxAngle
+        this@Roulette.setOnTouchListener(RouletteTouchListener())
+      }
 
-    selectedPosition = rTop
-  }
+      override fun onAnimationStart(p0: Animation?) {
 
-  /**
-   * Set the wheel image.
+      }
 
-   * @param drawableId the id of the drawable to be used as the wheel image.
-   */
-  fun setWheelImage(drawableId: Int) {
-    setImageResource(drawableId)
+    })
 
   }
 
-  /**
-   * get the angle of a touch event.
-   */
-  private fun getAngle(x: Double, y: Double): Double {
-    var x = x
-    var y = y
-    x -= rWidth / 2.0
-    y = rHeight.toDouble() - y - rHeight / 2.0
-
-    when (getQuadrant(x, y)) {
-      1 -> return Math.asin(y / Math.hypot(x, y)) * 180 / Math.PI
-      2 -> return 180 - Math.asin(y / Math.hypot(x, y)) * 180 / Math.PI
-      3 -> return 180 + -1.0 * Math.asin(y / Math.hypot(x, y)) * 180.0 / Math.PI
-      4 -> return 360 + Math.asin(y / Math.hypot(x, y)) * 180 / Math.PI
-      else -> return 0.0
-    }
+  private fun setQuadrant(angle: Float) {
+    current = if (angle < 72)  "Hardware"
+    else if (angle < 144) "Network"
+    else if (angle < 216)  "Software"
+    else if (angle < 288)  "Enterprise"
+    else "History"
   }
 
-  /**
-   * get the quadrant of the wheel which contains the touch point (x,y)
-
-   * @return quadrant 1,2,3 or 4
-   */
-  private fun getQuadrant(x: Double, y: Double): Int {
-    if (x >= 0) {
-      return if (y >= 0) 1 else 4
-    } else {
-      return if (y >= 0) 2 else 3
+  private fun generateRotationAnge(): Float {
+    when ((Math.random() * 5).toInt()) {
+      0 -> return 72.0f
+      1 -> return 144.0f
+      2 -> return 216.0f
+      3 -> return 288.0f
     }
+    return 360.0f
   }
 
-  /**
-   * rotate the wheel by the given angle
-
-   * @param degrees
-   */
-  private fun rotateWheel(degrees: Float) {
-    val animation: Animation = RotateAnimation(0.0f, degrees,
-        Animation.RELATIVE_TO_SELF,0.5f,
-        Animation.RELATIVE_TO_SELF,0.5f)
-    animation.repeatCount = 0
-    animation.interpolator = RouletteInterpolator()
-    animation.duration = 3000
-    animation.fillAfter = true
-    super.startAnimation(animation)
-    rotation += degrees
-
-  }
-
-  /**
-   * Interface to to observe user selection changes.
-   */
   interface RouletteChangeListener {
-    /**
-     * Called when user selects a new position in the wheel menu.
-
-     * @param selectedPosition the new position selected.
-     */
-    fun onSelectionChange(selectedPosition: Int)
+    fun onSelectionChange(selectedPosition: String)
```

Comment:
Here we could do an observable and notify other objects when the roulette has changed.